packages/mongo-nav/src/on-element-click-provider/OnElementClick.spec.tsx

```diff
@@ -0,0 +1,273 @@
+  [NavElement.ProjectNavInvite]: 'project-nav-invite',
+  [NavElement.ProjectNavActivityFeed]: 'project-nav-activity-feed',
+  [NavElement.ProjectNavAlerts]: 'project-nav-alerts',
+  [NavElement.UserMenuTrigger]: 'user-menu-trigger',
+};
+
+const orgSelectLink = {
+  [NavElement.OrgNavViewAllOrganizations]: 'org-select-view-all-orgs',
+};
+
+const orgDropdownLinks = {
+  [NavElement.OrgNavDropdownOrgAccessManager]:
+    'org-nav-dropdown-org-access-manager',
+  [NavElement.OrgNavDropdownProjectAccessManager]:
+    'org-nav-dropdown-org-project-manager',
+};
+
+const projectSelectLink = {
+  [NavElement.ProjectNavViewAllProjects]: 'project-select-view-all-projects',
+  [NavElement.ProjectNavAddProject]: 'project-select-add-new-project',
+};
+
+const projectDropdownLinks = {
+  [NavElement.ProjectNavProjectSettings]: 'project-nav-settings',
+  [NavElement.ProjectNavProjectSupport]: 'project-nav-settings',
+  [NavElement.ProjectNavProjectIntegrations]: 'project-nav-integrations',
+};
+
+const userMenuLinks = {
+  [NavElement.UserMenuFeedback]: 'user-menuitem-feedback',
+  [NavElement.UserMenuCloudMFA]: 'user-menuitem-cloud-mfa',
+  [NavElement.UserMenuCloudInvitations]: 'user-menuitem-cloud-invitations',
+  [NavElement.UserMenuCloudOrganizations]: 'user-menuitem-cloud-organizations',
+  [NavElement.UserMenuCloudUserPreferences]:
+    'user-menuitem-cloud-user-preferences',
+  [NavElement.Logout]: 'user-menuitem-logout',
+};
+
+// types
+interface ExpectedElements {
+  [key: string]: nullableElement;
+}
+
+describe('packages/mongo-nav/on-element-click-provider', () => {
+  const queries: Queries = {};
+  const expectedElements: ExpectedElements = {};
+
+  const setQueries = ({ queryByTestId }: Queries) => {
+    Object.assign(queries, { queryByTestId });
+    act(setExpectedElements);
+  };
+
+  const setExpectedElements = () => {
+    const { queryByTestId = () => null } = queries;
+    expectedElements.test = null;
+    expectedElements[NavElement.Logout] = queryByTestId('user-menuitem-logout');
+    expectedElements[NavElement.OrgNavLeaf] = queryByTestId('org-nav-leaf');
+    expectedElements[NavElement.OrgNavOrgSelectTrigger] = queryByTestId(
+      'org-trigger',
+    );
+    expectedElements[NavElement.OrgNavViewAllOrganizations] = queryByTestId(
+      'org-select-view-all-orgs',
+    );
+    expectedElements[NavElement.OrgNavOrgSettings] = queryByTestId(
+      'org-trigger-settings',
+    );
+    expectedElements[NavElement.OrgNavAccessManager] = queryByTestId(
+      'org-nav-access-manager',
+    );
+    expectedElements[NavElement.OrgNavDropdown] = queryByTestId(
+      'org-nav-dropdown',
+    );
+    expectedElements[NavElement.OrgNavDropdownOrgAccessManager] = queryByTestId(
+      'org-nav-dropdown-org-access-manager',
+    );
+    expectedElements[
+      NavElement.OrgNavDropdownProjectAccessManager
+    ] = queryByTestId('org-nav-dropdown-project-access-manager');
+    expectedElements[NavElement.OrgNavSupport] = queryByTestId(
```

Comment:
This is a lot of repeat logic. Couldn't you do something like this?
```js
const expectedElements = Object.keys(NavElement).reduce((acc, val) => {
  acc[val] = queryByTestId(NavElement[val])
  return acc
}, {} as ExpectedElements)
```

Reply:
Had to rewrite test suite, still not super clean but lmk what you think now... significantly decreased the warnings in the console..

Reply:
Seems significantly cleaner than before 👍